earth_enterprise/rpms/post-install_server.sh

```diff
@@ -0,0 +1,324 @@
+FUSIONBININSTALL="$BININSTALLROOTDIR/gefusion"
+SOURCECODEDIR=$(dirname $(dirname $(readlink -f "$0")))
+TOPSOURCEDIR_EE=$(dirname $SOURCECODEDIR)
+GESERVERBININSTALL="$BININSTALLROOTDIR/geserver"
+GEHTTPD="$BASEINSTALLDIR_OPT/gehttpd"
+GEHTTPD_CONF="$GEHTTPD/conf.d"
+
+run_as_user() {
+    local use_su=`su $1 -c 'echo -n 1' 2> /dev/null  || echo -n 0`
+    if [ "$use_su" -eq 1 ] ; then
+        >&2 echo "cd / ;su $1 -c \"$2\""
+        ( cd / ;su $1 -c "$2" )
+    else
+        >&2 echo "cd / ;sudo -u $1 $2"
+        ( cd / ;sudo -u $1 $2 )
+    fi
+}
+
+#-----------------------------------------------------------------
+
+# config values
+PUBLISHER_ROOT="/gevol/published_dbs"
+DEFAULTGROUPNAME="gegroup"
+GROUPNAME=$DEFAULTGROUPNAME
+DEFAULTGEFUSIONUSER_NAME="gefusionuser"
+GEFUSIONUSER_NAME=$DEFAULTGEFUSIONUSER_NAME
+
+# user names
+GRPNAME="gegroup"
+GEPGUSER_NAME="gepguser"
+GEAPACHEUSER_NAME="geapacheuser"
+
+SERVER_INSTALL_OR_UPGRADE="install"
+GEE_CHECK_CONFIG_SCRIPT="/opt/google/gehttpd/cgi-bin/set_geecheck_config.py"
+PGSQL_DATA="/var/opt/google/pgsql/data"
+PGSQL_LOGS="/var/opt/google/pgsql/logs"
+PGSQL_PROGRAM="/opt/google/bin/pg_ctl"
+PRODUCT_NAME="$GEE"
+
+#-----------------------------------------------------------------
+# Main Functions
+#-----------------------------------------------------------------
+
+main_postinstall()
+{
+  # 1) Modify files
+  modify_files
+
+  # 2) Set Permissions Before Server Start/Stop
+  fix_postinstall_filepermissions
+
+  # 3) PostGres DB config
+  postgres_db_config
+
+  # 4) Creating Daemon thread for geserver
+  setup_geserver_daemon
+
+  # 5) Add the server to services
+  # For RedHat and SuSE
+  test -f $CHKCONFIG && $CHKCONFIG --add geserver
+
+  # 6) Register publish root
+  $BASEINSTALLDIR_OPT/bin/geconfigurepublishroot --noprompt --path=$PUBLISHER_ROOT
+
+  # 7) Install the GEPlaces and SearchExample Databases
+  install_search_databases
+
+  # 8) Set permissions after geserver Start/Stop.
+  fix_postinstall_filepermissions
+
+  # TODO - verify
+  # 9) Run geecheck config script
+  # If file ‘/opt/google/gehttpd/cgi-bin/set_geecheck_config.py’ exists:
+  if [ -f "$GEE_CHECK_CONFIG_SCRIPT" ]; then
+    cd $BASEINSTALLDIR_OPT/gehttpd/cgi-bin
+    python ./set_geecheck_config.py
+  fi
+
+  # 10)
+  show_final_success_message
+}
+
+#-----------------------------------------------------------------
+# Post-install Functions
+#-----------------------------------------------------------------
+
+postgres_db_config()
+{
+  # a) PostGres folders and configuration
+  chmod -R 700 /var/opt/google/pgsql/
+  chown -R $GEPGUSER_NAME:$GRPNAME /var/opt/google/pgsql/
+  reset_pgdb
+}
+
+reset_pgdb()
+{
+  # TODO check if correct
+  # a) Always do an upgrade of the psql db
+  echo 2 | run_as_user $GEPGUSER_NAME "/opt/google/bin/geresetpgdb upgrade"
+  echo -e "upgrade done"
+
+  # b) Check for Success of PostGresDb
+  #  If file ‘/var/opt/google/pgsql/data’ doesn’t exist:
```

Comment:
This looks like it needs to have some code to check for success, instead of a comment.

Reply:
Already there - just look down a few lines.

Reply:
You're right. I was a bit confused by this stand-alone comment.